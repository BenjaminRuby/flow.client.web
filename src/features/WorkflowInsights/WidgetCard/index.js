import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import "./styles.scss";

const WidgetCard = ({ title, type, children }) => {
  return (
    <div className={classnames("b-widget-card", { [`--${type}`]: type })}>
      <div className="b-widget-card__title">{title}</div>
      <div className="b-widget-card__content">{children}</div>
    </div>
  );
};

WidgetCard.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.oneOf(["stat", "graph"]),
  children: PropTypes.any.isRequired
};

export default WidgetCard;
