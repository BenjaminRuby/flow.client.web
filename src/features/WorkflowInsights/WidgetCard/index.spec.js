import React from "react";
import WidgetCard from "./index";

const title = "test title";
const children = "test children";

describe("WidgetCard --- Snapshot", () => {
  it("Capturing Snapshot of WidgetCard", () => {
    const { baseElement } = rtlRouterRender(<WidgetCard title={title} children={children} />);
    expect(baseElement).toMatchSnapshot();
  });
});
