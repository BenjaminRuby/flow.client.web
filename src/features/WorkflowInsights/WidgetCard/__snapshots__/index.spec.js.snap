// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetCard --- Snapshot Capturing Snapshot of WidgetCard 1`] = `
<body>
  <div>
    <div
      class="b-widget-card"
    >
      <div
        class="b-widget-card__title"
      >
        test title
      </div>
      <div
        class="b-widget-card__content"
      >
        test children
      </div>
    </div>
  </div>
</body>
`;
